mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RoutingKeyConverter.java

```diff
@@ -0,0 +1,69 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+class RoutingKeyConverter {
+    private static final String SEPARATOR = ":";
+
+    static String toRoutingKey(RegistrationKey key) {
+        return key.getClass().getName() + SEPARATOR + key.asString();
+    }
+
+    @VisibleForTesting
+    static RoutingKeyConverter forFactories(RegistrationKey.Factory... factories) {
+        return new RoutingKeyConverter(ImmutableSet.copyOf(factories));
+    }
+
+    private final Set<RegistrationKey.Factory> factories;
+
+    @Inject
+    RoutingKeyConverter(Set<RegistrationKey.Factory> factories) {
+        this.factories = factories;
+    }
+
+    RegistrationKey toRegistrationKey(String routingKey) {
+        return toRegistrationKey(Splitter.on(SEPARATOR).splitToList(routingKey));
```

Comment:
Maybe using the `on(Pattern)` method help you by splitting only the first occurrence of `String:`
then you will not need to join in order to recreate the value.